playbooks/roles/ocp-cost-management/tasks/main.yml

```diff
@@ -0,0 +1,126 @@
+---
+# tasks file for playbooks/roles/ocp-cost-management
+
+- name: Check if cluster operators and nodes are healthy
+  include_role:
+    name: check-cluster-health
+
+- name: Clean up
+  include_tasks: "{{ role_path }}/tasks/cmo-cleanup.yml"
+  when: cost_management_cleanup
+
+- name: cost management operator setup
+  include_tasks: "{{ role_path }}/tasks/cmo-setup.yml"
+  when: cost_management_setup
+
+- name: cost management operator test
+  include_tasks: "{{ role_path }}/tasks/cmo-test.yml"
+  when: cost_management_test
+
+- name: Create working directory
+  file:
+    state: "directory"
+    path: "{{ cost_management_directory }}"
+
+- name: Setup custom catalog source
+  block:
+  - name: Include the global-secret-update role
+    include_role:
+        name: global-secret-update
+
+  - name: Set fact variable for CatalogSource name
+    set_fact:
+      cost_management_catalogsource: "qe-app-registry"
+
+  - name: Include role to create ImageContentSourcePolicy and CatalogSource
+    include_role:
+      name: set-custom-catalogsource
+    vars:
+      custom_catalogsource_name: "{{ cost_management_catalogsource }}"
+      custom_catalogsource_display_name: "Cost Management Operator"
+      custom_catalogsource_image: "{{ cost_management_catalogsource_image }}"
+  when: cost_management_catalogsource_image != '' and cost_management_catalogsource_image != None
+
+- name: Search for the StorageClass
+  kubernetes.core.k8s_info:
+    kind: StorageClass
+  register: storage_class
+
+- name: Check for default StorageClass annotations
+  set_fact:
+    default_storage_class: "{{ storage_class.resources[0].metadata.annotations }}"
+  when:
+  - storage_class.resources|length != 0 and storage_class.resources[0].metadata.annotations is defined
+  - "'storageclass.kubernetes.io/is-default-class' in storage_class.resources[0].metadata.annotations"
+
+- name: Fail if default StorageClass does not exist
+  fail:
+    msg: 'Default StorageClass does not exist'
+  when: >
+    (default_storage_class is undefined) or
+    (default_storage_class is defined and not default_storage_class["storageclass.kubernetes.io/is-default-class"]|bool)
+
+- name: Deploy Cost Management Operator
+  block:
```

Comment:
use block statements only when conditional execution of the tasks is required

Reply:
done